tests/test_mcp_servers.py

```diff
@@ -1,100 +1,373 @@
             Whenever a tool is called, be sure return the Response in a friendly and helpful tone.
-            When you are asked to search the web you must use a tool. keep answer concise
-            """ ,
-            tools=[mcp_server_name],
-            tool_config={"tool_choice":"auto"},
-            sampling_params={"max_tokens":4096}
+            When you are asked to search the web you must use a tool. Keep answers concise.
+            """
+
+    # Create agent with tools
+    agent = Agent(
+        client,
+        model=model,
+        instructions=instructions,
+        tools=[toolgroup_id],
+        tool_config={"tool_choice": "auto"},
+        sampling_params={"max_tokens": max_tokens}
+    )
+
+    # Create session
+    session_id = agent.create_session(session_name=f"Test_{toolgroup_id}_{int(time.time())}")
+
+    # Execute query
+    turn_response = agent.create_turn(
+        messages=[
+            {
+                "role": "user",
+                "content": prompt
+            }
+        ],
+        session_id=session_id,
+        stream=False
+    )
+
+    return turn_response
+
+
+def add_metric(
+    server_type: str,
+    model: str,
+    query_id: str,
+    status: str,
+    tool_call_match: bool,
+    inference_not_empty: bool,
+    error: str = ""
+):
+    """Add a metric record to the CSV file."""
+    # Ensure results directory exists
+    results_dir = Path("results")
+    results_dir.mkdir(exist_ok=True)
+
+    # Use a fixed CSV file name
+    metrics_file = results_dir / "metrics.csv"
+
+    # Create file with headers if it doesn't exist
+    if not metrics_file.exists():
+        with open(metrics_file, 'w', newline='') as f:
+            writer = csv.writer(f)
+            writer.writerow([
+                'timestamp',
+                'server_type',
+                'model',
+                'query_id',
+                'status',
+                'tool_call_match',
+                'inference_not_empty',
+                'error'
+            ])
+
+    # Append the new metric
+    with open(metrics_file, 'a', newline='') as f:
+        writer = csv.writer(f)
+        writer.writerow([
+            datetime.now().strftime("%Y-%m-%d %H:%M:%S"),
+            server_type,
+            model,
+            query_id,
+            status,
+            tool_call_match,
+            inference_not_empty,
+            error
+        ])
+
+
+def run_test(server_type, model, query_obj, llama_client, logger):
+    """Run a single test for a specific server type, model, and query."""
+    server_configs = get_server_configs()
+    config = server_configs[server_type]
+    query_id = get_query_id(query_obj)
+    prompt = query_obj['query']
+    expected_tool_call = query_obj['tool_call']
+    tool_call_match = False
+    inference_not_empty = False
+
+    # Skip if MCP URL is not set
+    if not config["mcp_url"]:
+        logger.info(f"Skipping {server_type}: MCP URL not set")
```

Comment:
Why do I need to include the MCP URL? This should default to whats available on the llama stack server, right?